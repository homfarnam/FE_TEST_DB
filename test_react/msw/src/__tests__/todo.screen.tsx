import {
  render,
  screen,
  waitForElementToBeRemoved,
} from '@testing-library/react'
import { QueryClient, QueryClientProvider } from 'react-query'
import { MemoryRouter, Route, Switch } from 'react-router-dom'
import TodoScreen from '../screens/todo.screen'

const client = new QueryClient({
  defaultOptions: { queries: { retry: 0 } },
})

const renderTodoScreen = (pathname: string) => {
  return render(
    <MemoryRouter initialEntries={[{ pathname }]}>
      <QueryClientProvider client={client}>
        <Switch>
          <Route component={TodoScreen} path="/todos/:id" />
        </Switch>
      </QueryClientProvider>
    </MemoryRouter>
  )
}

test('render <TodoScreen />, load data successfully', async () => {
  const { container } = renderTodoScreen('/todos/1')
  expect(screen.getByText(/Loading/i)).toBeInTheDocument()
  await waitForElementToBeRemoved(screen.getByText(/Loading/i))
  expect(container).toMatchSnapshot()
})

test('render <TodoScreen />, load data failed', async () => {
  // there is no todo with an id of 2000 at the mocking service
  const { container } = renderTodoScreen('/todos/2000')
  expect(screen.getByText(/Loading/i)).toBeInTheDocument()
  await waitForElementToBeRemoved(screen.getByText(/Loading/i))
  expect(container).toMatchSnapshot()
})
