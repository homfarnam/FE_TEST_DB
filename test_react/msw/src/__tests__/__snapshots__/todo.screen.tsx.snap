// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <TodoScreen />, load data failed 1`] = `
<div>
  <header>
    <a
      href="/"
    >
      Home
    </a>
  </header>
  <div
    class="ErrorMsg___StyledDiv-sc-11bc7qw-0 cjOpSh"
  >
    Failed to load todo
  </div>
</div>
`;

exports[`render <TodoScreen />, load data successfully 1`] = `
<div>
  <header>
    <a
      href="/"
    >
      Home
    </a>
  </header>
  <div
    class="Todo___StyledDiv-sc-5e6tjb-0 cmvIFh"
  >
    <h4>
      delectus aut autem
    </h4>
    <p>
      Completed: 
      No
       
    </p>
  </div>
</div>
`;
